### DISCUSSION TITLE
window transparency on wayland

### DISCUSSION THREAD

**Discussion by PhoenixClank** [qtile/qtile#4511] [Q&A] [answered]
When setting a window's opacity to anything other than `1.0`, qtile prints a warning that transparency isn't supported by wlroots yet. The window's opacity does not change.

Next to the code that prints the warning is a comment that links to an issue in wlroots' gitlab, which has been closed 2 months ago. The issue itself links to a merge request that has been merged 4 months ago.

Window transparency used to work flawlessly with the previous versions of wlroots and qtile.

What happened in the meantime that caused this to not work anymore?

**Comment by m-col** [comment, marked as answer]
We moved to a different wlroots API for rendering and organising window surfaces. The API simplifies things a lot on our end, but previously didn't implement server-side transparency if I remember correctly. If a client sets its own transparency then it will work, but Qtile can't then apply a level of transparency too it too.

**Reply by m-col** [reply]
To add to this, this isn't possible in the current release of wlroots. However, it should be possible in the next release of wlroots, and therefore for us in the Qtile release following wlroots' next release.

**Reply by PhoenixClank** [reply]
Why wasn't I notified of your answer? Thanks for the explanation! I'll be waiting patiently, then.

**Reply by jwijenbergh** [reply]
It's also already possible with #4524

**Comment by PhoenixClank** [comment]
> To add to this, this isn't possible in the current release of wlroots. However, it should be possible in the next release of wlroots, and therefore for us in the Qtile release following wlroots' next release.

There's been a new release of wlroots (0.17) a while ago, and a new release of qtile (0.24) recently. However, (at least on Arch) qtile still depends on wlroots 0.16. Window transparency still doesn't work.

Is compositor-set window transparency possible in wlroots 0.17? Will it be supported in qtile 0.25, whenever that releases?

And sorry if I sound impatient, I'm really not. I've started being used to my windows having frames again.
